src/java.desktop/windows/native/libawt/windows/awt_Window.cpp

```diff
@@ -3240,12 +3299,24 @@ void AwtWindow::_SetFullScreenExclusiveModeState(void *param)
     jboolean state = data->isFSEMState;
 
     PDATA pData;
-    JNI_CHECK_PEER_GOTO(self, ret);
+    if (self == NULL) {
+        env->ExceptionClear();
+        JNU_ThrowNullPointerException(env, "self");
+        delete data;
+        return;
+    } else {
+        pData = JNI_GET_PDATA(self);
```

Comment:
set directly

Reply:
@TheShermanTanker What this means is that you need to replace the line:
```
    PDATA pData;
```
above with:
```
 AwtWindow *window;
```
 
and remove 
```
    AwtWindow *window = (AwtWindow *)pData;
```
below, and then change all references to `pData` in this block to `window`.

The same changes needs to be done in all places where pData is references. This was a temporary variable that is no longer needed.